### DISCUSSION TITLE
How do I add a development-mode only server middleware with `nuxt/kit`?

### DISCUSSION THREAD

**Discussion by Robjam** [nuxt/nuxt#25247] [Questions] [answered]
Some background: I am working on a module to get CloudFlare's local environment (miniflare) to work with `nuxt dev`. Looking over the Nuxt docs it seems like Nuxt/Kit's [`addServerHandler`](https://nuxt.com/docs/api/kit/nitro#addserverhandler) would be the last piece of the puzzle to add `cloudflare` bindings to the context via the `middleware?: boolean` parameter. Unfortunately, `middleware` doesn't exist anymore. Any hints to getting a middleware working from nuxt/kit's setup method? Here's some example code of what I am trying:

```ts
// modules/nuxt-miniflare.ts

export default defineNuxtModule({
  // ...meta & defaults omitted
  setup(miniflareOptions) {
      if (!nuxt.options.dev) {
        return
      }
      let mf: Miniflare
      nuxt.hooks.hookOnce('build:before', async () => {
        const mf = new Miniflare({
          modules: true,
          // other configurations(bindings, etc.) omitted
          // an entrypoint is required for miniflare constructor
          script: `
            export default() {
              async fetch(request, context, env) {
                return new Resposnse('Hello from Miniflare!')
              }
            }
          `,
        })
        await mf.ready
        const bindings = await getBindings()
        addDevServerHandler({
          handler: (event) => {
          // this is never output to stdout, so it doesn't seem like it is called
          console.log('injecting cloudflare context');
          (event.context as any).cloudflare = env: {
            ...bindings,
          }
        }
    })
  }
})
```
I feel like I am so close to getting this working, any help would be greatly appreciated.

**Comment by Robjam** [comment]
It looks like someone else has the same issue in nitro?
https://github.com/unjs/nitro/issues/2044

**Comment by huang-julien** [comment]
Are you looking for `addDevServerHandler()` ?

It accept an object with route and handler. `route` can be omitted to be a middleware

**Reply by Robjam** [reply]
Thanks for taking a look! So the middleware flag isn't needed if `route` is not set. I am using `addDevServerHandler` without a route though and the middleware is not being called:
```ts
        addDevServerHandler({
          handler: (event) => {
          // this is never output to stdout, so it doesn't seem like it is called
          console.log('injecting cloudflare context');
          (event.context as any).cloudflare = env: {
            ...bindings,
          }
        }
```
Just a thought, is `addDevServerHandler` being added after the application middleware routes? I'll take a look later and see if that is why I don't see the logs!

**Comment by Robjam** [comment, marked as answer]
Miniflare/Wrangler added a new feature called `getPlatformProxy` that enables attaching a debugger after installing https://github.com/pi0/nitro-cloudflare-dev
This works for my use-case of locally debugging server calls that use R2, D1 and KV.

Awesome work pi0!
